.ci/jenkins/Jenkinsfile.deploy

```diff
@@ -121,28 +121,34 @@ pipeline {
         stage('Build OptaPlanner') {
             steps {
                 script {
-                    getOptaplannerMavenCommand().skipTests(params.SKIP_TESTS).run('clean install')
+                    getOptaplannerMavenCommand().withProperty('maven.test.failure.ignore', true).skipTests(params.SKIP_TESTS).run('clean install')
```

Comment:
Why are we ignoring test failures in the maven commands throughout this PR?

Reply:
we are ignoring the test failures here because we don't want the maven command to fail on test failure.
the [junit jenkins method](https://github.com/kiegroup/optaplanner/pull/1715/files#diff-794236ffded09cf2fd0d453b1f490e7c0f86b0c24426860e93c8ac9df67dd0d8R133) will set the result as unstable if there are any test failures.

so that we know better:
- Maven build failure -> compilation failure or any problem when building -> result of job will be FAILURE
- Maven test failure -> maven command is ok but junit will parse the result and set the job as UNSTABLE